internal/pkg/pkg.go

```diff
@@ -36,9 +36,22 @@ const ParentDir = ".."
 
 // Pkg represents a kpt package with a one-to-one mapping to a directory on the local filesystem.
 type Pkg struct {
-	UniquePath  types.UniquePath
+	// UniquePath represents absolute unique OS-defined path to the package directory on the filesystem.
+	UniquePath types.UniquePath
+
+	// DisplayPath represents Slash-separated path to the package directory on the filesystem relative
+	// to parent directory of root package on which the command is invoked.
+	// root package is defined as the package on which the command is invoked by user
+	// This is not guaranteed to be unique (e.g. in presence of symlinks) and should only
+	// be used for display purposes and is subject to change.
 	DisplayPath types.DisplayPath
 
+	// rootPkgParentDirPath is the absolute path to the parent directory of root package
+	// root package is defined as the package on which the command is invoked by user
+	// this must be same for all the nested subpackages in root package
+	// UniquePath = filepath.Join(rootPkgParentDirPath, DisplayPath)
+	rootPkgParentDirPath string
```

Comment:
Should avoid raw string types. Should this be a UniqePath?

Reply:
UniquePath has a different semantic. It is absolute path to the package. rootPkgParentDirPath is the absolute path to the parent dir of the root package on which command is invoked. It is private variable and not exposed, so left it to be a string.